src/wit_gen.rs

```diff
@@ -62,6 +66,27 @@ impl<F> TraceContext<F> {
 
         self.witness.step_instances.push(witness);
     }
+
+    pub fn set_num_steps(&mut self, num_steps: usize) {
```

Comment:
@lancenonce we should not have this method, num_steps should be passed to the constructor of TraceContext